proposed/http-client/http-client.md

```diff
@@ -12,8 +12,12 @@ interpreted as described in [RFC 2119](http://tools.ietf.org/html/rfc2119).
 ### Client
 
 An HTTP client has the responsibility to send a PSR-7 request and return a PSR-7
-response. The HTTP client MAY modify the response from the server before returning 
-the PSR-7 response. 
+response. The HTTP client MAY modify the request, given by the user and sent to 
```

Comment:
"Under the hood the HTTP client MAY modify the request/response received from the user/server."

How does that sound?

This part tries to make it clear that under the contract surface anything could happen to request/response objects and there is no assurance that they will remain the same objects on the instance level, right?